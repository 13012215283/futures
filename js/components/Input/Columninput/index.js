import React, { Component } from 'react';
import { View, TextInput, DeviceEventEmitter } from 'react-native';
import PropTypes from 'prop-types';

import styles from './style';
import { colors } from '../../themes';

export default class Columninput extends Component {
  static defaultProps = {
    placeholder: '请输入值',
  };

  constructor(props) {
    super(props);

    this.state = {
      text: '',
    };

    this.changeText = this.changeText.bind(this);
    this.getValue = this.getValue.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  onChange = () => {
    DeviceEventEmitter.emit('inputTextIsNull');
  };

  getValue = () => this.state.text;

  changeText(text) {
    this.setState({ text });
  }

  render() {
    const { placeholder } = this.props;

    return (
      <View style={styles.inputRectangle}>
        <TextInput
          style={styles.inputValue}
          underlineColorAndroid="transparent"
          placeholder={placeholder}
          placeholderTextColor={colors['1103']}
          onChangeText={this.changeText}
          onChange={this.onChange}
        />
      </View>
    );
  }
}

Columninput.propTypes = {
  placeholder: PropTypes.string,
};
