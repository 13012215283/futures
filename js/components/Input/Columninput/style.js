import { StyleSheet } from 'react-native';

import { colors } from '../../themes';

const styles = StyleSheet.create({
  inputRectangle: {
    height: 50,
    flexDirection: 'column',
    borderBottomColor: colors['1104'],
    borderBottomWidth: 1,
  },
  inputValue: {
    paddingLeft: 5,
    paddingBottom: 3,
  },
});

export default styles;
