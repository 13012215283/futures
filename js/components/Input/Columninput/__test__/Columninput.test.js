import React from 'react';

import renderer from 'react-test-renderer';
import Columninput from '../index';

/** 身份验证中的输入框
 * @property placeholder 输入框的提示语 default:'请输入'
 */

it('renders correctly', () => {
  const tree = renderer
    .create(<Columninput placeholder="请输入姓名" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
