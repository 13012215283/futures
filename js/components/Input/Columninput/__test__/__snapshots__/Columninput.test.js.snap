// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#e0e0e0",
      "borderBottomWidth": 1,
      "flexDirection": "column",
      "height": 50,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChange={[Function]}
    onChangeText={[Function]}
    placeholder="请输入姓名"
    placeholderTextColor="#bdbdbd"
    style={
      Object {
        "paddingBottom": 3,
        "paddingLeft": 5,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
